zetaclient/bitcoin_client_test.go

```diff
@@ -26,3 +31,44 @@ func TestConfirmationThreshold(t *testing.T) {
 		require.Equal(t, int64(bigValueConfirmationCount), client.ConfirmationsThreshold(big.NewInt(1000)))
 	})
 }
+
+func TestAvgFeeRateBlock828440(t *testing.T) {
+	// get archived block 828440
+	var blockVb btcjson.GetBlockVerboseTxResult
+	err := testutils.LoadObjectFromJSONFile(&blockVb, path.Join(testutils.TestDataPath, "bitcoin_block_828440.json"))
+	require.NoError(t, err)
+
+	// https://mempool.space/block/000000000000000000025ca01d2c1094b8fd3bacc5468cc3193ced6a14618c27
+	var blockMb testutils.MempoolBlock
+	err = testutils.LoadObjectFromJSONFile(&blockMb, path.Join(testutils.TestDataPath, "mempool.space_block_828440.json"))
+	require.NoError(t, err)
+
+	gasRate, err := CalcBlockAvgFeeRate(&blockVb, &chaincfg.MainNetParams)
+	require.NoError(t, err)
+	require.Equal(t, int64(blockMb.Extras.AvgFeeRate), gasRate)
+}
+
+func TestCalcDepositorFee828440(t *testing.T) {
```

Comment:
We should get bigger coverage for this functions. Testing error cases.

Reply:
I think you meant testing the error paths of this `CalcBlockAvgFeeRate` func.

Reply:
Yes

Reply:
[commit](https://github.com/zeta-chain/node/pull/1747/commits/489fe5bb288e3c8f6d0b55092f3195a19507fea7)